lib/puppet/provider/user/useradd.rb

```diff
@@ -55,35 +55,44 @@ def uid
      get(:uid)
   end
 
+  def comment
+     return localcomment if @resource.forcelocal?
+     get(:comment)
+  end
+
   def finduser(key, value)
     passwd_file = "/etc/passwd"
-    passwd_keys = ['account', 'password', 'uid', 'gid', 'gecos', 'directory', 'shell']
+    passwd_keys = [:account, :password, :uid, :gid, :gecos, :directory, :shell]
     index = passwd_keys.index(key)
     File.open(passwd_file) do |f|
       f.each_line do |line|
-         user = line.split(":")
-         if user[index] == value
-             f.close
-             return user
-         end
+        user = line.split(":")
+        if user[index] == value
+          return Hash[passwd_keys.zip(user)]
+        end
```

Comment:
The old and new code read `/etc/passwd` once per user resource per managed parameter. Can we not do that (in a future ticket/PR)? Seems like we could read `/etc/passwd` once per `user` resource without too much difficulty. Reading it once per transaction is also a possibility but users could be created mid-transaction...

Reply:
Good catch. Filed [PUP-10117](https://tickets.puppetlabs.com/browse/PUP-10117) for this issue.